lib/utils/fs.go

```diff
@@ -241,13 +242,53 @@ func FSTryReadLockTimeout(ctx context.Context, filePath string, timeout time.Dur
 	return fileLock.Unlock, nil
 }
 
+// RemoveAllSecure is similar to [os.RemoveAll] but leverages [RemoveSecure] to delete files.
+func RemoveAllSecure(path string) error {
+	// Match os.RemoveAll protections in not permitting removal of "." directories
+	if path == "." || (len(path) >= 2 && path[len(path)-1] == '.' && os.IsPathSeparator(path[len(path)-2])) {
+		return &os.PathError{Op: "RemoveAllSecure", Path: path, Err: syscall.EINVAL} // error type matches os.RemoveAll
+	}
+
+	if info, err := os.Lstat(path); err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return trace.ConvertSystemError(err)
+	} else if !info.IsDir() {
+		// file or symlink
+		return RemoveSecure(path)
+	}
+	var removeErrors []error
+	files, err := os.ReadDir(path)
+	if err != nil {
+		// don't fail fast, allow removal at end to be attempted
+		removeErrors = []error{err}
+	}
+	// it's possible for a partial file list to be returned even if an error above was returned
```

Comment:
```suggestion
	// It's possible for a partial file list to be returned even if an error above was returned.
```

https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences